### DISCUSSION TITLE
Implement detection of `image/x-icon` for `next/image` component

### DISCUSSION THREAD

**Discussion by kaguya3222** [vercel/next.js#45998] [Ideas]
### Describe the feature you'd like to request

Currently, Next.js tries to load icon with `fetch` request and then it determines it's content-type by following steps:
1. Tries to analyze response buffer and infer content-type. Currently 'analyzer' supports following content-types: `image/avif`, `image/webp`, `image/png`, `image/jpeg`, `image/gif`, `image/svg+xml`. 
2. Tries to get content type from request headers, like `response.headers.get('Content-Type').

So, I think it would be great if Next.js could infer `image/x-icon` content type from response buffer. 

### Describe the solution you'd like

I want Next.js to automatically detect `image/x-icon` content type of loaded image, even if response doesn't have correct `Content-Type` header.

### Describe alternatives you've considered

I think the only alternative is to have a correct response `Content-Type` header from URL where the loaded image is located. 
But, I think it will be more convenient if Next.js can automatically detect it, like it does for png, avif, webp and etc

**Comment by styfle** [comment]
The `image/x-icon` format is not currently optimized, its just a pass-through as seen here:

https://github.com/vercel/next.js/blob/fe762d329fdf9d6d6e6bf084198ad49f696d42b7/test/integration/image-optimizer/test/util.ts#L269-L289

When there is no upstream `content-type` header, the type is autodetected here:

https://github.com/vercel/next.js/blob/aeef38e80e8a5d5b90a0c290f9d1b64d62ff9846/packages/next/src/server/image-optimizer.ts#L112-L144

Feel free to submit a PR with a test to check for the magic number of ico 👍

**Reply by kaguya3222** [reply]
@styfle Thank you for your feedback! I will work on this
